pkg/schemaconv/smd.go

```diff
@@ -116,12 +140,72 @@ func (c *convert) makeRef(model proto.Schema) schema.TypeRef {
 
 		if tr == (schema.TypeRef{}) {
 			// emit warning?
-			tr.Inlined.Untyped = &schema.Untyped{}
+			tr.NamedType = &untypedName
 		}
 	}
 	return tr
 }
 
+func makeUnion(extensions map[string]interface{}) (*schema.Union, error) {
+	union := schema.Union{
+		Fields: []schema.UnionField{},
+	}
+
+	if idiscriminator, ok := extensions["x-kubernetes-union-discriminator"]; ok {
+		discriminator, ok := idiscriminator.(string)
+		if !ok {
+			return nil, fmt.Errorf(`"x-kubernetes-union-discriminator" is of invalid type: %#v`, idiscriminator)
```

Comment:
nit: write which type you expect

Reply:
Very good point, I just fixed them all! Thanks, PTAL